url9d32402837_dusan_rychnovsky_chess_endgames_tgz-pJ8-cz_dusanrychnovsky_chessendgames_core_strategies_PrecomputedValuesJ8/src/cz/dusanrychnovsky/chessendgames/core/Column.java

```diff
@@ -0,0 +1,228 @@
+package cz.dusanrychnovsky.chessendgames.core;
+
+import java.util.Iterator;
+
+/**
+ * 
+ * @author Dušan Rychnovský
+ *
+ */
+public enum Column
+{
+	CA,
+	CB,
+	CC,
+	CD,
+	CE,
+	CF,
+	CG,
+	CH;
+	
+	private static final Column[] values = values();
+	
+	/**
+	 * 
+	 * @param ordinal
+	 * @return
+	 */
+	public static Column get(int ordinal)
+	{
+		try {
+			return values[ordinal];
+		}
+		catch (ArrayIndexOutOfBoundsException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)